platform/engineering/design-docs/2020-12-10-auto-deploy-cms-build.md

```diff
@@ -0,0 +1,216 @@
+- The daily production deploy of `vets-website`, which will build with latest release of `vets-website`
+- The `va-cms-bot`, which will build with the latest commit on `vets-website` in response to a manual trigger on the part of the CMS team
+
+#### Pain points
+
+- Content writers want to quickly draft, publish, and deploy content.
+  - However, `vets-website` is only deployed to production once per day.
+  - The CMS team can run content-only deploys independently of the daily production deploy, but that is triggered manually.
+- The content-only deploy does not include automated accessibility checks.
+- In local development, engineers can encounter confusing build errors that are due to the mismatch in outdated locally cached content and updated Liquid templates.
+
+### High Level Design
+
+This new auto-deploy has a few main differences with the existing content-only deploy.
+- Uses the CMS export to build content
+- Runs on a schedule instead of being triggered by other jobs (such as the daily prod deploy)
+- Deploys to production instead of only dev and staging
+
+The schedule will run between 8am and 8pm ET. It will initially be set to run hourly as a conservative cadence. It will then be incrementally adjusted to a higher frequency as the build system allows.
+
+To avoid deploying when content hasn't changed since the previous deploy, the job will compare the export data (e.g., using and archiving the checksums of the tarballs).
+
+## Specifics
+
+### Detailed Design
+
+#### Creating a new pipeline
+
+A new pipeline (e.g., `Jenkinsfile.autodeploy`) will be created for the auto-deploy. It will be modeled after the current `Jenkinsfile.content` configuration.
+
+#### Using the CMS export to build content
+
+In order for the partial deploy to use the CMS export data, the [build command](https://github.com/department-of-veterans-affairs/vets-website/blob/master/jenkins/common.groovy#L189) in the `build` function in `jenkins/common.groovy` will have to include the `--use-cms-export` flag.
+
+Keep in mind that this content-only deploy will build using the latest release instead of the latest commit of of `vets-website`. This matches the behavior of the current content-only deploy. The daily production deploy will continue deploying from the latest commit and retain the responsibility of tagging releases.
+
+If this work is done before the full build has transitioned to using the CMS export, the inclusion of the flag will be contingent on the `contentOnly` parameter to the function.
+
+#### Validating the build
+
+The implementation of the full build pipeline may introduce new stages or common helper functions that the auto-deploy pipeline can incorporate. Specifically, the full build pipeline will compare the diffs between GraphQL and CMS export builds. That validation should be done in this pipeline as well while this process is being rolled out.
```

Comment:
@indytechcook the diff comparison will only be added to the build for the purposes of release. Once CMS Export is rolled out, the assumption is that it will be removed